main.cpp

```diff
@@ -402,6 +444,63 @@ IntType& IntType::divide(int rhs)
     return *this;
 }
 
+struct Point
+{
+    Point(const FloatType& fta, const FloatType& ftb)
+    {
+        FloatType xCopy = const_cast<FloatType&>(fta);
+        x = static_cast<float>(xCopy);
+        FloatType yCopy = const_cast<FloatType&>(ftb);
+        y = static_cast<float>(yCopy);
+    }
+    Point(const DoubleType& dta, const DoubleType& dtb)
+    {
+        x = static_cast<float>( const_cast<DoubleType&>(dta));
+        y = static_cast<float>( const_cast<DoubleType&>(dtb));
+    }
+    Point(const IntType& ita, const IntType& itb)
+    {
+        x = const_cast<IntType&>(ita);
+        y = const_cast<IntType&>(itb);
+    }
+    Point(const Point& other)
+    {
+        x = other.x;
+        y = other.y;
+    }
+    Point& multiply(float m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point& multiply(FloatType m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point& multiply(DoubleType m)
+    {
+        x *= static_cast<float>(m);
+        y *= static_cast<float>(m);
+        return *this;
+    }
+    Point& multiply(IntType m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
```

Comment:
You are passing your custom types by-copy here.  Pass by const-reference.
all of these multiply functions should call `multiply(float)` internally and return what that function returns.
Remember: Don't Repeat Yourself.   You repeated `x *= m; y *= m;` more than once.  That means you should refactor and D.R.Y.
- [ ] revise.